api/unstable/fingerprint.h

```diff
@@ -74,3 +74,28 @@ int s2n_client_hello_get_fingerprint_hash(struct s2n_client_hello *ch,
 int s2n_client_hello_get_fingerprint_string(struct s2n_client_hello *ch,
         s2n_fingerprint_type type, uint32_t max_size,
         uint8_t *output, uint32_t *output_size);
+
+/**
+ * Creates an s2n_client_hello from bytes representing a ClientHello message.
+ *
+ * Unlike s2n_connection_get_client_hello, the s2n_client_hello produced by this
+ * method is owned by the application and must be freed with s2n_client_hello_free.
+ *
+ * This method does not support SSLv2 ClientHellos.
+ *
+ * @param bytes The raw bytes representing the ClientHello.
+ * @param size The size of raw_message.
+ * @returns A new s2n_client_hello on success, or NULL on failure.
+ */
+struct s2n_client_hello *s2n_client_hello_parse_message(uint8_t *bytes, uint32_t size);
```

Comment:
No. The version is included in the ClientHello.

Even if we did want to support SSLv3 ClientHellos, the version would be included in the ClientHello. We just don't support that because of how s2n-tls implemented SSLv3 ClientHello parsing.

Reply:
Hmm or maybe the version wouldn't be distinguishable from the SSLv2 ClientHello 🤔 The distinction between an SSLv2 ClientHello message and record isn't as clear as later versions. You can guess a record is an SSLv2 ClientHello, but if you strip off the record header you can't. 

But since we don't support SSLv2 and both the record header protocol version and ClientHello protocol version are now frozen for future protocol versions, I don't think we need a version parameter.